mdpow/workflows/dihedrals.py

```diff
@@ -531,19 +536,29 @@ def dihedral_violins(df, width=0.9, solvents=SOLVENTS_DEFAULT, plot_title=None):
                     len(df[df['interaction'] == "VDW"]["lambda"].unique()),
                     len(df[df['interaction'] == "Coulomb"]["lambda"].unique()) - 1]
 
-    solvs = list(solvents)
-    if len(solvs) < 2:
-        solvs.append('N/A')
-    
-    g = sns.catplot(data=df, x="lambda", y="dihedral", hue="solvent", col="interaction",
-                    kind="violin", split=True, width=width, inner=None, cut=0,
-                    linewidth=0.5,
-                    hue_order=[solvs[0], solvs[1]], col_order=["Coulomb", "VDW", "Structure"],
-                    sharex=False, sharey=True,
-                    height=3.0, aspect=2.0,
-                    facet_kws={'ylim': (-180, 180),
-                               'gridspec_kws': {'width_ratios': width_ratios,
-                                                }})
+    # Usage in Jupyter causes matplotlib figure object output, not the completed figure
+    # Upcoming fix in issue #260
+    assert 0 < len(solvents) < 3, "one or two solvents must be specified, otherwise SOLVENTS_DEFAULT is used"
+    if len(list(solvents)) < 2: 
+        g = sns.catplot(data=df, x="lambda", y="dihedral", hue="solvent", col="interaction",
+                        kind="violin", split=False, width=width, inner=None, cut=0,
+                        linewidth=0.5,
+                        hue_order=list(solvents), col_order=["Coulomb", "VDW", "Structure"],
+                        sharex=False, sharey=True,
+                        height=3.0, aspect=2.0,
+                        facet_kws={'ylim': (-180, 180),
+                                'gridspec_kws': {'width_ratios': width_ratios,
+                                                    }})
+    else:
+        g = sns.catplot(data=df, x="lambda", y="dihedral", hue="solvent", col="interaction",
+                        kind="violin", split=True, width=width, inner=None, cut=0,
+                        linewidth=0.5,
+                        hue_order=list(solvents), col_order=["Coulomb", "VDW", "Structure"],
+                        sharex=False, sharey=True,
+                        height=3.0, aspect=2.0,
+                        facet_kws={'ylim': (-180, 180),
+                                'gridspec_kws': {'width_ratios': width_ratios,
+                                                    }})
```

Comment:
That's a bit clunky and has a lot of repeated code. I suggest to unify (then you also don't have to test extra lines to make codecov happy... although in principle tests for 1 and 2 solvents _would_ be good)
```suggestion
    split = len(solvents) > 1
    g = sns.catplot(data=df, x="lambda", y="dihedral", hue="solvent", col="interaction",
                    kind="violin", split=split, width=width, inner=None, cut=0,
                    linewidth=0.5,
                    hue_order=list(solvents), col_order=["Coulomb", "VDW", "Structure"],
                    sharex=False, sharey=True,
                    height=3.0, aspect=2.0,
                    facet_kws={'ylim': (-180, 180),
                               'gridspec_kws': {'width_ratios': width_ratios}}
                    )

```
I think the two invocations only differed by `split` but if there are other things then I'd change the parameters and then feed the parameters to _one_ call to `catplot`.

Reply:
@orbeckst 
Hey Oliver, I made these changes. I also added a test, but our data in the testing resources only contain single solvent analyses and results, so it only tests the single solvent usage. I think that is okay for now, since the `assert` exists in the plotting code, and the only other option with two solvents is `split=True`.